cli/pkg/kctrl/cmd/package/repository/add_or_update.go

```diff
@@ -257,6 +260,10 @@ func (o *AddOrUpdateOptions) updateExistingPackageRepository(pkgr *kcpkg.Package
 		ImgpkgBundle: &kappctrl.AppFetchImgpkgBundle{Image: o.URL},
 	}
 
+	if strings.TrimSpace(o.Secret) != "" {
```

Comment:
Do we need TrimSpace? i think cobra CLI will take care of Trimming Spaces.

Reply:
@rohitagg2020 Removed the check.
URL check with empty quoted strings (" ")  returns a OCI parse error.
Do we need to do something like that here or this should fine for current scope?